### PR TITLE
Java 17 cleanup

### PR DESCRIPTION
**Pull request by Aurelien30000**
## Description

This PR handles the migration/cleanup to Java 17 and precedes #173.

## Submitter Checklist

- [x] Make sure you are opening from a topic branch (**/feature/fix/docs/ branch** (right side)) and not your main branch.
- [x] Ensure that the pull request title represents the desired changelog entry.
- [x] New public fields and methods are annotated with `@since TODO`.
- [x] I read and followed the [contribution guidelines](https://github.com/IntellectualSites/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Aurelien30000** [comment]
@NotMyFault Is something wrong with this pr? Sorry to ping you, but I would like to be able to finish implementing the _really interesting_ pr (#173) that now requires this one ;) Thanks!

**Comment by NotMyFault** [comment]
> @NotMyFault Is something wrong with this pr? Sorry to ping you, but I would like to be able to finish implementing the _really interesting_ pr (#173) that now requires this one ;) Thanks!

No, I was waiting for you to strip the cleanup from the other PR to facilitate a quicker review, but that didn't happen yet, so I kept postponing this PR in favor of other PRs.

**Comment by Aurelien30000** [comment]
Woops, I will add the rest then.

**Comment by Aurelien30000** [comment]
Ready, hopefully I didn't neglect anything 😅

**Comment by Aurelien30000** [comment]
@NotMyFault I added some other extra cleanup _(method refactoring)_ with ``sendInfo``. You can review this PR then, thanks!

**Comment by NotMyFault** [comment]
Conflicts

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -1,194 +1,194 @@
 # The config-version line is immutable. Don't change the value, it will reset itself after every reboot.
-config-version: 3
```
Comment (NotMyFault) [inline src/main/resources/config.yml]: Unneeded change

Comment (Aurelien30000) [reply src/main/resources/config.yml]: Oops 😅

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**